### PR TITLE
Add easyconfigs for ncurses when building with gcc 5

### PR DESCRIPTION
**Pull request by akesandgren**
Also included patch from ncurses upstream to fix the actual problem.

### CODE REVIEW

File: easybuild/easyconfigs/n/ncurses/ncurses-6.0_gcc-5.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
+diff --git a/ncurses/base/MKlib_gen.sh b/ncurses/base/MKlib_gen.sh
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-6.0_gcc-5.patch]: @akesandgren please include a reference to where you got this patch, and a short description of what it fixes (any lines above the `---` line are comment, so just add some lines at the very top)

File: easybuild/easyconfigs/n/ncurses/ncurses-6.0-GCCcore-5.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ and more. It uses Terminfo format, supports pads and color and multiple highlights and forms characters and
+ function-key mapping, and has all the other SYSV-curses enhancements over BSD Curses."""
+
+toolchain = {'name': 'GCCcore', 'version': '5.3.0'}
+toolchainopts = {'optarch': True, 'pic': True}
+
+source_urls = [GNU_SOURCE]
+sources = [SOURCE_TAR_GZ]
+
+patches = ['ncurses-6.0_gcc-5.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-6.0-GCCcore-5.3.0.eb]: @akesandgren please use `%(version)s` rather than hardcoding `6.0`

**Comment by akesandgren** [comment]
commit amended with changes according to your request.

**Comment by boegel** [comment]
@akesandgren missing dep:

```
EasyBuildError: 'Irresolvable dependencies encountered: GNU/5.3.0-2.26'
```

**Comment by akesandgren** [comment]
Dropped the ncurses-6.0-GNU-5.3.0-2.26.eb since there currently is no GNU-5.3.0-2.26 compiler toolchain defined.
Commit amended and should hopefully work now.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9e45a4bc219fd320dbea39b8ef4a9f2b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2459.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/8430268310b202b0793f7908f8698abf for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @akesandgren!

**Pull request merged**